Therapeut Vechter/Assets/Scripts/LevelScreen/Old Spawning/SpawnLevels.cs

```diff
@@ -0,0 +1,97 @@
+using System.Linq;
+using UnityEngine;
+
+namespace LevelScreen
+{
+    public class SpawnLevels : MonoBehaviour
+    {
+        [SerializeField] private GameObject[] blockPrefabs;
+        public int SpawnDistance;
+        [SerializeField] private int startDistance = 25;
+        [SerializeField] private int lineWidth;
+        [SerializeField] private LineRenderer lineRenderComplete;
+        private bool lineCompleteDrawn;
+        private int lastCompleteInt;
+        [SerializeField] private LineRenderer lineRenderUnComplete;
+        private int lineUnCompleteInt;
+        [SerializeField] private LineRenderer lineRenderGradient;
+        private int lineGradientInt;
+        private bool lineGradientDrawn;
+        private bool allLinesDrawn;
+        public Transform[] levelPrefabPosition;
+
+        private void Start()
+        {
+            //set line render width
+            lineRenderComplete.startWidth = lineWidth;
+            lineRenderComplete.sortingOrder = -1;
+            lineRenderUnComplete.startWidth = lineWidth;
+            lineRenderUnComplete.sortingOrder = -1;
+            lineRenderGradient.startWidth = lineWidth;
+            lineRenderGradient.sortingOrder = -1;
+            //spawn level objects in
+            blockPrefabs = Resources.LoadAll<GameObject>("Prefabs");
+            for (var i = 0; i < blockPrefabs.Length; i++)
+            {
+                var levelPrefab = Instantiate(blockPrefabs[i], new Vector3(startDistance + i * SpawnDistance, 0, 0),
+                    Quaternion.identity, transform);
+            }
+        }
+
+        //line renderer functionality
+        private void Update()

```

Comment:
move function out of update into its own named function and then put that function in the update fucntion